common/thorhelper/thorsoapcall.cpp

```diff
@@ -2243,5 +2322,7 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
 
 IWSCAsyncFor * createWSCAsyncFor(CWSCHelper * _master, IXmlWriterExt &_xmlWriter, ConstPointerArray &_inputRows, PTreeReaderOptions _options)
 {
+    if (!persistentInitDone)
+        initPersistentHandler();
```

Comment:
this double check pattern isn't 100% safe as it stands, persisentInitDone needs to be an atomic.
See : https://preshing.com/20130930/double-checked-locking-is-fixed-in-cpp11/